src/Operation/Factory/SubresourceOperationFactory.php

```diff
@@ -183,7 +178,11 @@ private function computeSubresourceOperations(string $resourceClass, array &$tre
 
             $tree[$operation['route_name']] = $operation;
 
-            $this->computeSubresourceOperations($subresourceClass, $tree, $rootResourceClass, $operation, $visited + [$visiting => true], ++$depth, $maxDepth);
+            // get the minimum maxDepth from the rootMaxDepth and the maxDepth of the to be visited Subresource
+            $currentMaxDepth = array_diff([$maxDepth, $subresource->getMaxDepth()], [null]);
```

Comment:
optionally array_filter([$maxDepth, $subresource->getMaxDepth()], 'is_int')

Reply:
Indeed, is way more performant thx